ern-core/src/FsCache.ts

```diff
@@ -0,0 +1,220 @@
+    }
+    if (!fs.existsSync(this.cacheManifestPath)) {
+      writeJSONSync(this.cacheManifestPath, { entries: [], size: 0 })
+    }
+  }
+
+  /**
+   * Indicates whether a given object is currently stored in the cache
+   * @param obj The object to check for presence in the cache
+   */
+  public async isInCache(obj: T): Promise<boolean> {
+    const manifestObj = await this.getManifestObj()
+    return !!manifestObj.entries.find(e => e.id === this.objectToId(obj))
+  }
+
+  /**
+   * Adds an object entry to the cache
+   * @param obj The object to add to the cache
+   */
+  public async addToCache(obj: T): Promise<string> {
+    if (await this.isInCache(obj)) {
+      throw new Error(
+        `Object with id ${this.objectToId(obj)} is already in the cache`
+      )
+    }
+    // Create unique directory to store the object
+    const pathToObjectCacheDir = this.createUniqueCacheDirectory()
+    // Invoke client provided function to write object in cache directory
+    await this.addObjectToDirectory(obj, pathToObjectCacheDir)
+    // Compute cached object directory size
+    const cachedObjectDirSize = await getSizeAsync(pathToObjectCacheDir)
+    // Create manifest entry for new cached object
+    const entry: CacheManifestEntry = {
+      id: this.objectToId(obj),
+      lastAccessed: Date.now(),
+      path: pathToObjectCacheDir,
+      size: cachedObjectDirSize,
+    }
+    // Update manifest total cache size
+    const manifestObj = await this.getManifestObj()
+    manifestObj.size += cachedObjectDirSize
+    manifestObj.entries.push(entry)
+    // Clean cache if newly added object caused total cache
+    // size to exceed the defined max cache size
+    // Remove all entries that were least recently accessed
+    // until the total cache size falls below the max cache size
+    const entriesSortedByLat = manifestObj.entries.sort(
+      (a, b) => b.lastAccessed - a.lastAccessed
+    )
+    while (manifestObj.size >= this.maxCacheSize) {
+      const entryToRemove = entriesSortedByLat.pop()
+      manifestObj.size -= entryToRemove!.size
+      shell.rm('-rf', entryToRemove!.id)
+    }
+    manifestObj.entries = entriesSortedByLat
+    // Save updated manifest
+    await writeJSON(this.cacheManifestPath, manifestObj)
+    return pathToObjectCacheDir
+  }
+
+  /**
+   * Gets path to the directory containing a cached object
+   * @param obj The object for which to retrieve cached directory path
+   */
+  public async getObjectCachePath(obj: T): Promise<string | void> {
+    const manifestObj = await this.getManifestObj()
+    const cacheEntry = manifestObj.entries.find(
+      e => e.id === this.objectToId(obj)
+    )
+    if (cacheEntry) {
+      cacheEntry.lastAccessed = Date.now()
+      await writeJSON(this.cacheManifestPath, manifestObj)
+      return cacheEntry.path
+    }
+  }
+
+  /**
+   * Gets the cache manifest object
+   */
+  private async getManifestObj(): Promise<CacheManifest> {
+    return readJSON(this.cacheManifestPath)
```

Comment:
readJSONSync()?

Reply:
This is an `async` method, it returns a `Promise`. `readJson` returns a Promise so nothing wrong here.